plugins/MosaicCamera/src/MosaicCamera.hpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2024 Josh Meyers
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA  02110-1335, USA.
+ */
+
+#ifndef MOSAICCAMERA_HPP
+#define MOSAICCAMERA_HPP
+
+#include "StelGui.hpp"
+#include "StelModule.hpp"
+#include "StelPluginInterface.hpp"
+
+class StelButton;
+class MosaicCameraDialog;
+
+/*! @defgroup mosaicCamera Mosaic Camera plug-in
+@{
+The Mosaic Camera plugin overlays camera sensor boundaries on the sky.
+@}
+*/
+
+/// @brief Represents a set of polygons with associated properties.
+struct PolygonSet
+{
+    QString name;                      ///< The name of the polygon set.
+    QVector<QVector<QPointF>> corners; ///< Polygons as vectors of corner points.
+    QColor color;                      ///< Color associated with the polygon set.
+};
+
+
+/// @brief Represents a camera with its properties and associated polygon sets.
+struct Camera
+{
+    QString name;                    ///< The name of the camera.
+    double ra;                       ///< Right Ascension of camera pointing [deg]
+    double dec;                      ///< Declination of camera pointing [deg]
+    double rotation;                 ///< Rotation angle of the camera [deg]
+    bool visible;                    ///< Visibility status of the camera.
+    QVector<PolygonSet> polygon_sets; ///< Collection of polygon sets associated with the camera.
+};
+
+//! @class MosaicCamera
+//! @ingroup mosaicCamera
+//! Main class of the Mosaic Camera plug-in.
+//! @author Josh Meyers
+class MosaicCamera : public StelModule
+{
+	Q_OBJECT
+    /**
+     * @name Camera properties
+     * @{
+     * We maintain the concept of a "current" camera primarily for the convenience of
+	 * setting and getting properties through the Stellarium Remove Control HTTP API.
+     */
+
+	/// @property enabled
+	/// @brief Are mosaic camera overlays enabled?
+	Q_PROPERTY(bool enabled		     READ isEnabled              WRITE enableMosaicCamera  NOTIFY flagMosaicCameraVisibilityChanged)
+
+	Q_PROPERTY(bool showButton       READ getFlagShowButton      WRITE setFlagShowButton   NOTIFY flagShowButtonChanged)
+
+	/// @property currentCamera
```

Comment:
The StelProperties fully work only when NOTIFY signals are used. Please add them like 

`... NOTIFY currentRAChanged)`